metric-anomaly-detector/src/test/java/org/hypertrace/alert/engine/metric/anomaly/detector/evaluator/MetricCacheTest.java

```diff
@@ -0,0 +1,117 @@
+                    List.of(
+                        new StringJoiner(".")
+                            .add(AttributeScope.SERVICE.name())
+                            .add("startTime")
+                            .toString(),
+                        new StringJoiner(".")
+                            .add(AttributeScope.SERVICE.name())
+                            .add("duration")
+                            .toString()),
+                    new String[][] {
+                      {String.valueOf(Duration.ofMinutes(1).toMillis()), "100"},
+                      {String.valueOf(Duration.ofMinutes(2).toMillis()), "100"},
+                      {String.valueOf(Duration.ofMinutes(3).toMillis()), "100"},
+                      {String.valueOf(Duration.ofMinutes(4).toMillis()), "100"},
+                      {String.valueOf(Duration.ofMinutes(5).toMillis()), "100"},
+                    }))
+            .iterator();
+
+    Mockito.when(
+            queryRequestHandler.executeQuery(anyMap(), any(), anyString(), anyLong(), anyLong()))
+        .thenReturn(itr);
+
+    MetricCache metricCache = new MetricCache(queryRequestHandler);
+
+    LhsExpression lhsExpression = createLhsExpression("name", AttributeScope.SERVICE.name());
+    RhsExpression rhsExpression = createRhsExpression("customer");
+    LeafFilter leafFilter =
+        createLeafFilter(ValueOperator.VALUE_OPERATOR_EQ, lhsExpression, rhsExpression);
+    MetricSelection metricSelection =
+        MetricSelection.newBuilder()
+            .setMetricAggregationInterval("PT15s")
+            .setMetricAggregationFunction(
+                MetricAggregationFunction.METRIC_AGGREGATION_FUNCTION_TYPE_SUM)
+            .setFilter(Filter.newBuilder().setLeafFilter(leafFilter).build())
+            .setMetricAttribute(
+                Attribute.newBuilder().setKey("duration").setScope("SERVICE").build())
+            .build();
+
+    String tenantId = "tenant1";
+    long currentTime = Duration.ofMinutes(5).toMillis();
+    // loads data from pinot
+    List<Pair<Long, Double>> list =
+        metricCache.getMetricValues(
+            Map.of(),
+            metricSelection,
+            tenantId,
+            currentTime - Duration.ofMinutes(5).toMillis(),
+            currentTime);
+    Assertions.assertEquals(5, list.size());
+    MetricTimeSeries metricTimeSeries = metricCache.getMetricTimeSeriesRecord(metricSelection);
+    Assertions.assertEquals(5, metricTimeSeries.getDataList().size());
+
+    // returns data from cache
+    list =
+        metricCache.getMetricValues(
+            Map.of(),
+            metricSelection,
+            tenantId,
+            currentTime - Duration.ofMinutes(3).toMillis(),
+            currentTime);
+    Assertions.assertEquals(4, list.size());
+    metricTimeSeries = metricCache.getMetricTimeSeriesRecord(metricSelection);
+    Assertions.assertEquals(5, metricTimeSeries.getDataList().size());
+
+    // triggers trimming of data
```

Comment:
I think, we have some logic for managing this window for trimming. I have commented above, let me know what do you think? Can we make this simple or do we need this?

Reply:
replied